pkg/await/await.go

```diff
@@ -333,18 +347,34 @@ func Deletion(
 	if awaiter, exists := awaiters[id]; exists && awaiter.awaitDeletion != nil {
 		waitErr = awaiter.awaitDeletion(ctx, clientForResource, name)
 	} else {
-		objectMissing := func(d *unstructured.Unstructured, err error) error {
-			if is404(err) {
-				return nil
-			} else if err != nil {
-				glog.V(3).Infof("Received error deleting object '%s': %#v", d.GetName(), err)
-				return err
+		_ = host.LogStatus(ctx, diag.Info, urn, fmt.Sprintf("Waiting for deletion of %s '%s'", id, name))
+
+		for {
+			select {
+			case event, ok := <-watcher.ResultChan():
+				if !ok {
+					return fmt.Errorf("Timed out waiting for deletion of %s '%s'", id, name)
+				}
+
+				switch event.Type {
+				case watch.Deleted:
+					return nil
+				case watch.Error:
```

Comment:
What sorts of errors does `watch.Error` signify? I don't think we handle this case anywhere else, which might not be good. :)

Reply:
Still trying to figure out the answer to this one. So far, it looks like it may only happen for client errors like providing an invalid resource version to the watch API.

Reply:
I'm going to leave this as is for now. Ultimately, I'd like to migrate to using an informer rather than the low-level watch API, which will handle this detail. I don't think this will cause problems in the meantime.

Reply:
Sounds good.